src/components/YextEntityFieldSelector.tsx

```diff
@@ -42,33 +43,45 @@ export const YextEntityFieldSelector = <T extends Record<string, any>>(
     label: props.label,
     render: ({ field, value, onChange }: RenderProps) => {
       const filteredEntityFields = getFilteredEntityFields(props.filter);
+      const [entityFieldMode, setEntityFieldMode] = useState<boolean>(true);
+
+      const toggleEntityFieldMode = () => {
+        setEntityFieldMode((prevMode) => !prevMode);
+        onChange({
+          field: value?.field ?? "",
+          constantValue: value?.constantValue ?? "",
+          constantValueEnabled: !value.constantValueEnabled,
+        });
+      };
 
       return (
         <>
-          <FieldLabel label={field.label || "Label is undefined"}>
-            <AutoField
-              field={{
-                type: "select",
-                options: [
-                  { value: "", label: "Select a Content field" },
-                  ...filteredEntityFields.map((entityFieldNameToSchema) => {
-                    return {
-                      label: entityFieldNameToSchema.name,
-                      value: entityFieldNameToSchema.name,
-                    };
-                  }),
-                ],
-              }}
-              onChange={(selectedEntityField) => {
-                onChange({
-                  field: selectedEntityField,
-                  constantValue: value?.constantValue ?? "",
-                });
-              }}
-              value={value?.field}
-            />
-          </FieldLabel>
-          {shouldDisplayConstantValueField(props.filter.types) && (
+          {entityFieldMode ? (
+            <FieldLabel label={field.label || "Label is undefined"}>
+              <AutoField
+                field={{
+                  type: "select",
+                  options: [
+                    { value: "", label: "Select a Content field" },
+                    ...filteredEntityFields.map((entityFieldNameToSchema) => {
+                      return {
+                        label: entityFieldNameToSchema.name,
+                        value: entityFieldNameToSchema.name,
+                      };
+                    }),
+                  ],
+                }}
+                onChange={(selectedEntityField) => {
+                  onChange({
```

Comment:
Since you set all of the values here, I don't think `toggleEntityFieldMode` is necessary at all. The check on entityFieldMode on line 59 should be `!value?.constantValueEnabled`.

Reply:
When you click "Use Constant Value", the constantValueEnabled isn't changed immediately, but only onChange for the fieldValue. That's why I added toggleEntityFieldMode, so it is changed right away on toggle.

Reply:
Gotcha. But you don't need entityFieldMode